x/gamm/keeper/math.go

```diff
@@ -98,6 +98,30 @@ func calcOutGivenIn(
 	return tokenAmountOut
 }
 
+// CalcOutGivenIn calculates token to be provided, fee added,
+// given the swapped out amount, using solveConstantFunctionInvariant.
+// This version is exported and can be used to calculate the exact output of a swap.
+func CalcOutGivenIn(
```

Comment:
Is this being used somewhere? This should probably not be exposed for any txfee purpose. (Its also not compatible post-API refactor)

Reply:
Per call: Lets keep it, just switch to doing it off of the pool interface

Reply:
Forgot to resolve this – resolved in a commit a couple days ago!